test/wasm/assets/test.js

```diff
@@ -104,6 +113,22 @@ function evaluate(policy, input) {
     return { addr: resultAddr };
 }
 
+function decodeResultSet(policy, result) {
+
+    const rawAddr = policy.module.instance.exports.opa_json_dump(result.addr);
+    const buf = new Uint8Array(policy.memory.buffer);
+
+    // NOTE(tsandall): There must be a better way of doing this...
+    let s = '';
+    let idx = rawAddr;
+
+    while (buf[idx] != 0) {
+        s += String.fromCharCode(buf[idx++]);
+    }
```

Comment:
I think the only better thing would be if we knew the length so we could do something like: `const s = String.fromCharCode(...buf.slice(rawAddr, rawAddr+len))`, as is it would require iterating over the buffer to find the null terminator though so I'm not sure its actually any nicer.

Reply:
For now it's not much of an improvement. In the future we could return the length of the buffer than then perhaps we could JSON parse the Uint8 array?